File: leetcode_python/remove_duplicates_from_sorted_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        if (!head) {
            return head;
        }
        ListNode *previous = head, *current = head -> next;

        while(current) {
            if (current -> val == previous -> val) {
                current = current -> next;
                previous -> next = current;
            }
            else {
                previous = current;
                current = current -> next;
            }
        }

        return head;
    }
};

File: leetcode_python/remove_duplicates_from_sorted_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param head, a ListNode
    # @return a ListNode
    def deleteDuplicates(self, head):
        if not head:
            return head
        previous, current = head, head.next
        while (current):
            if current.val == previous.val:
                current = current.next
                previous.next = current
            else:
                previous, current = current, current.next
        return head
